src/openrct2/ride/Ride.cpp

```diff
@@ -2816,7 +2822,7 @@ static void ride_call_mechanic(Ride* ride, Peep* mechanic, int32_t forInspection
  */
 static void ride_call_closest_mechanic(Ride* ride)
 {
-    auto forInspection = (ride->lifecycle_flags & (RIDE_LIFECYCLE_BREAKDOWN_PENDING | RIDE_LIFECYCLE_BROKEN_DOWN)) == 0;
+    auto forInspection = (ride->lifecycle_flags & RIDE_LIFECYCLE_BROKEN_DOWN) == 0;
```

Comment:
This is a change from vanilla. Is it something we want to do make it that mechanics aren't called until the ride has actually broken down.

Reply:
I will reply to this even though I've already seen your comments below, just to explain why I made the change on this line: It is possible to reproduce #8145 using the "force breakdown" feature without this change. While a mechanic is heading for an inspection, force an ideally long-lasting (i.e. train stuck somewhere on track) pending breakdown and then pick up / put down the mechanic. The inspection logic will try to call them again for the inspection through this function, but since there is a pending breakdown now, it will be a "respond to breakdown call" even though the breakdown is still pending. The mechanic will then walk to the vehicle that has the pending breakdown even though it's not in the station.

So I tried to fix this bug by making sure the mechanic would not get into the "fixing" state unless the ride has already broken down. So while a breakdown is pending and not visible to the player, I try to keep them in the inspection-related states (HEADING instead of ANSWERING). This function was one of the two places I identified that would put a mechanic into a fixing-related state "too early", and my change should prevent that from happening.

Obviously there are other ways to prevent this from happening, like handling the "fixing" process in the station differently if the breakdown is still pending when the mechanic arrives. It would be interesting to see how vanilla RCT2 handles this, but booting Windows is a hassle and it has no "force breakdown" feature AFAIK so I'm not really sure how it could be tested.